pkg/proxy/request_test.go

```diff
@@ -429,3 +429,96 @@ func TestGetDuration(t *testing.T) {
 		assert.Equal(t, scenario.expected, actual)
 	}
 }
+
+func TestGetImage(t *testing.T) {
+	for _, ti := range []struct {
+		tag                   string
+		role                  string
+		imageName             string
+		imageTag              string
+		expectedImageResponse string
+		expectedTagResponse   string
+		expectError           bool
+	}{
+		{
+			tag:                   "valid master image",
+			role:                  "masterImage",
+			imageName:             "hellofresh/kangal-jmeter-master",
+			imageTag:              "latest",
+			expectedImageResponse: "hellofresh/kangal-jmeter-master",
+			expectedTagResponse:   "latest",
+			expectError:           false,
+		},
+		{
+			tag:                   "valid worker set",
+			role:                  "workerImage",
+			imageName:             "hellofresh/kangal-jmeter-worker",
+			imageTag:              "latest",
+			expectedImageResponse: "hellofresh/kangal-jmeter-worker",
+			expectedTagResponse:   "latest",
+			expectError:           false,
+		},
+		{
+			tag:                   "valid empty master image and tag",
+			role:                  "masterImage",
+			imageName:             "",
+			imageTag:              "",
+			expectedImageResponse: "",
+			expectedTagResponse:   "",
+			expectError:           false,
+		},
+		{
+			tag:                   "invalid master image name",
+			role:                  "masterImage",
+			imageName:             "hellofresh/kangal-jmeter-master",
+			imageTag:              "latest",
+			expectedImageResponse: "hellofresh/kangal-jmeter-worker",
+			expectedTagResponse:   "latest",
+			expectError:           true,
+		},
+		{
+			tag:                   "invalid worker image tag",
+			role:                  "workerImage",
+			imageName:             "hellofresh/kangal-jmeter-worker",
+			imageTag:              "1.0",
+			expectedImageResponse: "hellofresh/kangal-jmeter-worker",
+			expectedTagResponse:   "latest",
+			expectError:           true,
+		},
+	} {
+		t.Run(ti.tag, func(t *testing.T) {
+
+			image := apisLoadTestV1.ImageDetails{
+				Image: "",
+				Tag:   "",
+			}
+
+			err := error(nil)
+
+			sentImage := ti.imageName + ":" + ti.imageTag
```

Comment:
You could provide the complete image name in this test scenarios, it would help to check cases when the `strings.Contains(imageStr, ":")` is false and the default value should be used.

Reply:
Sorry, didn't mean to refer to the default value from the backend, just the default returned from `getImage` when `strings.Contains(imageStr, ":")` is false, which should be an empty `apisLoadTestV1.ImageDetails` struct

Reply:
Ok ok, I get it. I will add a condition on the test so if any of image name or image tag is empty, an empty `apisLoadTestV1.ImageDetails` will be returned

Reply:
a good condition to test would be providing at the request `someImage/withNoColon` this should also output an empty `apisLoadTestV1.ImageDetails` as `strings.Contains(imageStr, ":")` would be false :wink:

Reply:
Yep, and it did not work like that. Empty apisLoadTestV1.ImageDetails is the expected return for empty or not valid images.
I was not very happy with the way the image url string was being treated, so now I use the module `github.com/novln/docker-parser`. It is included in the vendor and go.mod.

The suggested test is implemented as well.

Reply:
I don't see this lib in the modules :thinking: 
If I may advise you, don't worry too much about this and go for the simpler approach.. 

I think you were doing great by enforcing a `imagename:tag` pattern, it's the most common representation of an image so it's a very reasonable decision, as long we're clearly informing the user what we expect as an input (which you are already doing it with the regex in the validation) :relieved: